Hyper/Controllers/SkyboxController.cs

```diff
@@ -0,0 +1,229 @@
+    };
+
+    private Phase _dayStart = new()
+    {
+        SkytopColor = new Vector4(0.34f, 0.55f, 0.85f, 1f), // https://www.wolframalpha.com/input?i=%23568CD8
+        HorizonColor = new Vector4(0.51f, 0.7f, 0.91f, 1.0f), // https://www.wolframalpha.com/input?i=%2382B2E8
+        SunglareColor = Vector4.One,
+        StarsVisibility = 0.05f,
+    };
+
+    private Phase _sunsetStart = new()
+    {
+        SkytopColor = new Vector4(0.08f, 0.16f, 0.32f, 1f), // https://www.wolframalpha.com/input?i=%23152852
+        HorizonColor = new Vector4(0.99f, 0.37f, 0.33f, 1f), // https://www.wolframalpha.com/input?i=%23FD5E53
+        SunglareColor = new Vector4(1f, 0.8f, 0.2f, 1f),
+        StarsVisibility = 0.2f,
+    };
+
+    private Phase _duskStart = new()
+    {
+        SkytopColor = new Vector4(0.19f, 0f, 0.43f, 1f), // https://www.wolframalpha.com/input?i=%2330016d
+        HorizonColor = new Vector4(0.05f, 0.05f, 0.24f, 1f), // https://www.wolframalpha.com/input?i=%09%230d0d3e
+        SunglareColor = new Vector4(0.82f, 0.59f, 0.56f, 1f),
+        StarsVisibility = 0.5f,
+    };
+
+    private Phase _nightStart = new()
+    {
+        SkytopColor = Vector4.Zero,
+        HorizonColor = Vector4.Zero,
+        SunglareColor = Vector4.Zero,
+        StarsVisibility = 1f
+    };
+
+    public SkyboxController(Scene scene, AbstractSkyboxShader skyboxShader, StandardModelShader modelShader, StandardObjectShader objectShader, Settings settings)
+    {
+        _scene = scene;
+        _skyboxShader = skyboxShader;
+        _modelShader = modelShader;
+        _objectShader = objectShader;
+        _skybox = new Skybox.Skybox(skyboxShader.GlobalScale);
+        _initialSunVector = -Vector3.UnitZ;
+        _initialMoonVector = -Vector3.UnitX;
+        _settings = settings;
+        _initTimeSeconds = _settings.TimeInSeconds;
+        _stopwatch.Start();
+    }
+
+    public void Dispose()
+    {
+        _settings.TimeInSeconds = GetCurrentTime();
+        _skyboxShader.Dispose();
+    }
+
+    // current time in seconds
+    private float GetCurrentTime()
+    {
+        float currentTime = _stopwatch.ElapsedMilliseconds / 1000.0f + _initTimeSeconds;
+        if (currentTime > _dayLengthSeconds * 2)
+        {
+            _stopwatch.Restart();
+            if (_initTimeSeconds != 0)
+                _initTimeSeconds = 0;
+        }
+        return currentTime;
+    }
+
+    private (Phase, Phase, float) GetInterval(float time)
+    {
+        float sunriseStartTime = 0f;
+        float dayStartTime = 0.1f * _dayLengthSeconds;
+        float dayEndTime = 0.9f * _dayLengthSeconds;
+        float sunsetStartTime = 1f * _dayLengthSeconds;
+        float duskStartTime = 1.15f * _dayLengthSeconds;
+        float nightStartTime = 1.3f * _dayLengthSeconds;
+        float nightEndTime = 1.7f * _dayLengthSeconds;
+        float dawnStartTime = 1.9f * _dayLengthSeconds;
+
+        if (time >= sunriseStartTime && time < dayStartTime)
+        {
```

Comment:
could this method be split into multiple? if not maybe remove the { } in ifs because this methods takes a lot of space

Reply:
removed { }